include/odp/api/spec/pool.h

```diff
@@ -251,6 +251,37 @@ typedef uintptr_t (*odp_pool_ext_mem_allocate_t)(uint32_t total_sz, unsigned int
  */
 typedef void (*odp_pool_ext_mem_deallocate_t)(uintptr_t ptr);
 
+/**
+ * ODP pool buffer type iterator callback.
+ *
+ * This callback shall be invoked by the implementation when each odp_buffer_t type is freed
+ * to the pool in odp_pool_create().
+ *
+ * @param data_addr Pointer to the buffer start address. On successful pool creation,
+ * the application can access the same buffer address using odp_buffer_addr().
+ *
+ */
+typedef void (*odp_pool_buffer_iterator_t)(void *data_addr);
```

Comment:
Some for of user context would be needed as well. Maybe a pointer to copy of pool parameters.

Reply:
Addressed in v3.